src/formatter.rs

```diff
@@ -10,44 +10,113 @@ lazy_static! {
+/// Others:
+///     https://www.codingame.com/contribute/view/1222536cec20519e1a630ecc8ada367dd708b
+fn format_edit_monospace(text: &str, ostyle: &OutputStyle) -> String {
+    if RE_MONOSPACE_OLD.is_match(&text) {
+        let msg = "Clash contains obsolete ``` formatting, consider fixing it in the website.";
+        println!("{} {}\n", ostyle.failure.paint("WARNING"), msg);
+    }
+
+    let mut result = text.replace("```", "`");
+
+    result = RE_MONOSPACE_TRIM
+        .replace_all(&result, |caps: &regex::Captures| format!("\n\n`{}`\n\n", &caps[1]))
+        .to_string();
+
+    result
+}
+
+/// If it's not inside a Monospace block, trim consecutive spaces.
+fn format_trim_consecutive_spaces(text: &str) -> String {
+    RE_BACKTICK
+        .replace_all(&text, |caps: &regex::Captures| {
+            if let Some(monospace_text) = caps.get(1) {
+                monospace_text.as_str().to_string()
+            } else if let Some(non_monospace_text) = caps.get(2) {
+                RE_SPACES.replace_all(non_monospace_text.as_str(), " ").to_string()
             } else {
                 "".to_string()
             }
         })
-        .to_string();
+        .to_string()
+}
 
-    // Make sure monospace blocks are on their own line, and remove extra
-    // whitespace around them
-    result = RE_MONOSPACE_TRIM
-        .replace_all(&result, |caps: &regex::Captures| format!("\n{}\n", &caps[1]))
-        .to_string();
+/// Adds padding to Monospace blocks.
+/// NOTE 1: Comes before the adding of reverse nester tags so that the Monospace
+///         block is not split into separate blocks and messes up the padding.
+/// NOTE 2: Needs to factor the fact that tags are going to be deleted later on.
+fn format_monospace_padding(text: &str) -> String {
+    RE_MONOSPACE
+        .replace_all(&text, |caps: &regex::Captures| {
+            let lines: Vec<&str> = caps[1].split('\n').collect();
+            let padding = lines.iter().map(|line| clean_line(line.to_string()).len()).max().unwrap_or(0);
+            let formatted_lines = lines
+                .iter()
+                .map(|&line| {
+                    let clean_line = clean_line(line.to_string());
+                    let offset = line.len() - clean_line.len();
+                    format!("{:<width$}", line, width = padding + offset)
+                })
+                .collect::<Vec<String>>()
+                .join("\n");
+            format!("`{}`", formatted_lines)
+        })
+        .to_string()
+}
 
-    // Nested tags (only some combinations).
-    // Hacky - it's based upon the fact that only 1-level nesting makes sense.
-    // Adds reverse nester brackets so that the following replacement logic will
-    // work. i.e : <<Next [[N]] {{3}} lines:>> becomes <<Next >>[[N]]<< {{3}}
-    // lines:>>
+/// Returns a line without formatting tags. Used for computing the padding.
+fn clean_line(line: String) -> String {
```

Comment:
I think this function could be replaced with `&str -> usize` function calculates the "real" length of a string (without formatting) since that's the only thing it's used for right now.